altair/vegalite/v5/api.py

```diff
@@ -3865,101 +3850,81 @@ def interactive(  # noqa: C901
+            legend_type = facet_legend["type"]
+            if utils.is_undefined(legend_type):
+                legend_type = facet_legend.to_dict(context={"data": chart.data})["type"]
+
+            if legend_type == "nominal":
+                # TODO Ideally this would work for ordinal data too
+                legend_selection = selection_point(bind="legend", encodings=[legend])
+                initial_computed_domain = param(expr=f"domain('{legend}')")
+                nonreactive_domain = param(
+                    react=False, expr=initial_computed_domain.name
                 )
-
-            if legend_encoding is not None:
-                if utils.is_undefined(
-                    interactive_chart.encoding[legend_encoding]["type"]
-                ):
-                    legend_encoding_type = interactive_chart.encoding[
-                        legend_encoding
-                    ].to_dict(context={"data": interactive_chart.data})["type"]
+                scale = facet_legend["scale"]
+                if utils.is_undefined(scale):
+                    scale = {"domain": nonreactive_domain}
                 else:
-                    legend_encoding_type = interactive_chart.encoding[legend_encoding][
-                        "type"
-                    ]
-                if (
-                    legend_encoding_type == "nominal"
-                ):  # TODO Ideally this would work for ordinal data too
-                    legend_selection = selection_point(
-                        bind="legend", encodings=[legend_encoding]
-                    )
-                    initial_computed_domain = param(expr=f"domain('{legend_encoding}')")
-                    nonreactive_domain = param(
-                        react=False, expr=initial_computed_domain.name
-                    )
-                    if utils.is_undefined(
-                        interactive_chart.encoding[legend_encoding]["scale"]
-                    ):
-                        interactive_chart.encoding[legend_encoding]["scale"] = {
-                            "domain": nonreactive_domain
-                        }
-                    else:
-                        interactive_chart.encoding[legend_encoding]["scale"][
-                            "domain"
-                        ] = nonreactive_domain
-
-                    interactive_chart = interactive_chart.add_params(
-                        legend_selection, initial_computed_domain, nonreactive_domain
-                    ).transform_filter(legend_selection)
-        return interactive_chart
+                    scale["domain"] = nonreactive_domain
+                chart = chart.add_params(
+                    legend_selection,
+                    initial_computed_domain,
+                    nonreactive_domain,
+                ).transform_filter(legend_selection)
+            else:
+                msg = f"Expected only 'nominal' legend type but got {legend_type!r}"
```

Comment:
Same reasoning as for [`a8001da` (#3394)](https://github.com/vega/altair/pull/3394/commits/a8001da0245ad5d7515c2a90877c7b7cb2b91ce0#r1712993209)

Reply:
Yes, that sounds good. I think the only other type we will support it `'ordinal'`. I don't see how either `'quantitative'` or `'temporal'` would make use of interactivity (I guess one could technically imagine an interval selection on a quantitative legend but that is currently not possible in VegaLite).